### PR TITLE
feat: Support MySQL CAST_DECIMAL function

### PR DESCRIPTION
**Pull request by robbinhan**
**What this PR does**:
Support MySQL CAST_DECIMAL function #571

### CODE REVIEW

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/function/cast_decimal.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		return nil, errors.WithStack(err)
+	}
+	v := math.ToDecimal(val)
+
+	maxNum, err := inputs[1].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if maxNum == nil {
+		maxNum = 0
```
Comment (Mulavar) [inline pkg/runtime/function/cast_decimal.go:64]: maybe the default max num is 10, refer to https://dev.mysql.com/doc/refman/5.6/en/fixed-point-types.html.

**Review by Mulavar — APPROVED** [review]

**Comment by Mulavar** [comment]
pls resolve the conflict.

File: pkg/runtime/function/cast_decimal.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+func (a castDecimalFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	if len(inputs) != 3 {
+		return "", errors.New("The Decimal function must accept three parameters\n")
+	}
+
+	val, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	v := math.ToDecimal(val)
```
Comment (jjeffcaii) [inline pkg/runtime/function/cast_decimal.go:57]: Use `val.Decimal()` instead.

File: pkg/runtime/function/cast_decimal.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	precision, err := inputs[2].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if precision == nil {
+		precision = 0
+	}
+	d, err := strconv.Atoi(fmt.Sprint(precision))
```
Comment (jjeffcaii) [inline pkg/runtime/function/cast_decimal.go:78]: Use `precision.Int64()` instead.

**Review by jjeffcaii — APPROVED** [review]

**Pull request merged**